ResearchUI/ResearchUI/iOS/Views/RSDStepChoiceCell.swift

```diff
@@ -344,23 +344,162 @@ open class RSDStepTextFieldCell: RSDTableViewCell {
+    override open var usesLightStyle: Bool {
+        didSet {
+            if usesLightStyle {
+                viewLabel.textColor = UIColor.rsd_textViewCellLabelLightStyle
+                textView.textColor = UIColor.rsd_textViewCellTextLightStyle
+                textView.layer.borderColor = UIColor.rsd_textViewCellBorderLightStyle.cgColor
+            }
+            else {
+                viewLabel.textColor = UIColor.rsd_textViewCellLabel
+                textView.textColor = UIColor.rsd_textViewCellText
+                textView.layer.borderColor = UIColor.rsd_textViewCellBorder.cgColor
+            }
+        }
+    }
+    
+    public override init(style: UITableViewCellStyle, reuseIdentifier: String?) {
+        super.init(style: style, reuseIdentifier: reuseIdentifier)
+        commonInit()
+    }
+    
+    required public init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+        commonInit()
+    }
+    
+    override open func awakeFromNib() {
+        super.awakeFromNib()
+        commonInit()
+    }
+    
+    func commonInit() {
+        
+        initializeViews()
+        
+        contentView.addSubview(textView)
+        contentView.addSubview(viewLabel)
+        
+        setupViews()
+        
+        textView.translatesAutoresizingMaskIntoConstraints = false
+        viewLabel.translatesAutoresizingMaskIntoConstraints = false
+        
+        setNeedsUpdateConstraints()
+    }
+    
+    override open func updateConstraints() {
+        
+        NSLayoutConstraint.deactivate(self.constraints)
+        
+        textView.rsd_removeSiblingAndAncestorConstraints()
+        
+        viewLabel.rsd_alignToSuperview([.leading, .trailing], padding: constants.sideMargin)
+        viewLabel.rsd_alignToSuperview([.top], padding: constants.verticalMargin)
+        
+        textView.rsd_alignToSuperview([.leading, .trailing], padding: constants.sideMargin)
+        textView.rsd_alignBelow(view: viewLabel, padding: constants.verticalPadding)
+        textView.rsd_makeHeight(.equal, constants.height)
+        
+        textView.rsd_alignToSuperview([.bottom], padding: constants.verticalMargin)
+        
+        super.updateConstraints()
+    }
+}
+
 /// `RSDStepTextFieldCellLayoutConstants` defines the layout constants used by a `RSDStepTextFieldCell`.
 public protocol RSDStepTextFieldCellLayoutConstants {
     var verticalMargin: CGFloat { get }
     var verticalPadding: CGFloat { get }
     var sideMargin: CGFloat { get }
 }
 
-/// Default constants.
+/// `RSDStepTextViewCellLayoutConstants` defines the layout constants used by a `RSDStepTextViewCell`.
+public protocol RSDStepTextViewCellLayoutConstants {
+    var verticalMargin: CGFloat { get }
+    var verticalPadding: CGFloat { get }
+    var sideMargin: CGFloat { get }
+    var height: CGFloat { get }
+    var textInset: CGFloat { get }
+    var borderRadius: CGFloat { get }
+}
+
+/// Default constants used by a `RSDStepTextFieldCell`.
```

Comment:
@syoung-smallwisdom  When we generate HTML API docs from the Swift doc comments, do you know if it includes documentation of non-public objects and interfaces like this (or if there's a command line switch to not do that)? It'd be nice to have them documented internally for framework developers, but if these would wind up in the external docs then they should just be plain old comments, not doc comments.

Reply:
No, right now it is set to only show doc comments for public. private, fileprivate, and internal are all hidden. That said, I will often use a comment on those functions on the assumption that a developer making changes may make them public in a future release.

Reply:
@syoung-smallwisdom so should I change "///" to "//" on these 'fileprivate struct' declarations?

Reply:
No, it's fine to use `///` and in fact, that's what I will use on the assumption that future development may call for changing the privacy.